app/src/main/java/org/mozilla/fenix/components/toolbar/BrowserToolbarView.kt

```diff
@@ -39,6 +46,35 @@ class BrowserToolbarView(
     val toolbarIntegration: ToolbarIntegration
 
     init {
+        view.setOnUrlLongClickListener {
+            val popup = PopupMenu(view.context, view)
+            popup.menuInflater.inflate(R.menu.browser_toolbar_popup_menu, popup.menu)
+            popup.show()
+
+            val clipboard = view.context.getSystemService(CLIPBOARD_SERVICE) as ClipboardManager
+
+            popup.menu.findItem(R.id.paste)?.isVisible = clipboard.containsText()
+            popup.menu.findItem(R.id.paste_and_go)?.isVisible = clipboard.containsText()
```

Comment:
Only display the "paste" or "paste & go" buttons if there is plain text available on the clipboard